api_types.ts

```diff
@@ -546,6 +546,36 @@ export interface CommonPackFormulaDef<T extends ParamDefs> {
    * @hidden
    */
   readonly allowedAuthenticationNames?: string[];
+
+  /**
+   * The JavaScript function that implements parameter validation.
+   *
+   * This function takes in parameters and a context containing a {@link PermissionSyncMode}
+   * and validates the parameters. A formula may want to validate parameters differently
+   * for permissionSyncMode 'PermissionAware' vs 'Personal' vs undefined (which represents a formula).
+   *
+   * @returns if the parameters are valid
+   * @throws {@link ParameterValidationError} if the parameters are invalid.
+   *
+   * @example
+   * ```
+   * validateParameters: async function (context, _, formulaContext) {
+   *   let [quantity, sku] = formulaContext?.params;
+   *   let errors = [];
+   *   if (quantity < 0) {
+   *     errors.push({message: "Must be a positive number.", propertyName: "quantity"});
+   *   }
+   *   if (!isValidSku(context, sku)) {
+   *     errors.push({message: `Product SKU not found.`, propertyName: "sku"});
+   *   }
+   *   if (errors.length > 0) {
+   *     throw new ParameterValidationError("Invalid parameter values.", errors);
+   *   }
+   * return true;
+   * },
+   * ```
+   */
+  validateParameters?: MetadataFormula<ExecutionContext, boolean>;
```

Comment:
For a sync table definition, do we need to override `ExecutionContext` with `SyncExecutionContext`?

Reply:
Yeah for precision is what I meant, it's actually important because our main use case is validating sync table and we probably need the extra bits from `SyncExecutionContext` to do some of our validation and we wouldn't want the packs to have to do a cast.